app/controllers/articles_controller.rb

```diff
@@ -2,38 +2,55 @@
 
 class ArticlesController < ApplicationController
   before_action :set_article, only: %i[show edit update destroy]
+  before_action :set_user, only: %i[show edit update destroy]
 
   def index
-    @articles = Article.all
+    @articles = policy_scope(Article)
   end
 
   def new
     @article = Article.new
   end
 
   def edit
-    @user = current_user
+    authorize @article
+  rescue Pundit::NotAuthorizedError
+    flash[:alert] = "Cannot access the article #{@article.title}"
+    redirect_to user_articles_path
   end
 
-  def show; end
+  def show
+    @article = policy_scope(Article).find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    flash[:alert] = "Cannot access the article #{@article.title}"
+    redirect_to user_articles_path
+  end
 
   def update
+    authorize @article
     if @article.update(article_params)
       flash[:success] = 'Article successfully updated'
       redirect_to user_article_path
     else
       flash.now[:danger] = 'Article update failed'
       render :edit
     end
+  rescue Pundit::NotAuthorizedError
```

Comment:
You can use ```rescue_from``` to create rescue for all ``` Pundit::NotAuthorizedError``` https://apidock.com/rails/ActiveSupport/Rescuable/ClassMethods/rescue_from